### PR TITLE
VIH-7879 Updated to use pagination for incrementing pages

### PR DESCRIPTION
**Pull request by PrasannaDommalapati**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-7879

### Change description ###

- valid base rule
- ELinksApiGetPeopleUpdatedSinceDays

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: SchedulerJobs/SchedulerJobs.AcceptanceTests/packages.lock.json [hunk 1]
```diff
@@ -119,8 +119,8 @@
       },
       "BookingsApi.Client": {
         "type": "Transitive",
-        "resolved": "1.25.3",
-        "contentHash": "KISOFCwReGSR0hkHRWtN1V5l8af40lAQ4h6E+4tJuXhLj5geylJVQ8zHzDmuMh0zz0tI6BKrJM31wa/rRVOiOw==",
+        "resolved": "1.27.1-beta0021",
```
Comment (FarronF) [inline SchedulerJobs/SchedulerJobs.AcceptanceTests/packages.lock.json:122]: Is this ok to use a beta version?

File: SchedulerJobs/SchedulerJobs.Services/ELinksService.cs [hunk 2]
```diff
@@ -31,11 +32,12 @@ public ELinksService(IPeoplesClient peoplesClient, ILeaversClient leaversClient,
 
         public async Task ImportJudiciaryPeopleAsync(DateTime fromDate)
         {
+            var invalidPeoplePersonalCode = new List<string>();
             var currentPage = 1;
 
-            try
+            while (true)
```
Comment (omio-ma) [inline SchedulerJobs/SchedulerJobs.Services/ELinksService.cs:38]: have we taken consideration about infinite loop here? Or are we confident that peopleresult list will eventually be zero? Although I can see this isn't a new change however we have changed the order of the try catch and while blocks here, does our unit test check of the list being null or empty as I cant see any other way this code will ever break?

File: SchedulerJobs/SchedulerJobs.Services/ELinksService.cs [hunk 3]
```diff
@@ -88,16 +93,17 @@ public async Task ImportLeaversJudiciaryPeopleAsync(DateTime fromDate)
...
                     currentPage++;
                 }
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError(ex, "There was a problem importing judiciary leavers");
-                throw;
+                catch (Exception ex)
+                {
+                    currentPage++;
```
Comment (omio-ma) [inline SchedulerJobs/SchedulerJobs.Services/ELinksService.cs:103]: Just double checking if this is intentional, so we are saying that if an exception is being thrown, we should still continue to be in an infinite loop here until leaversResult's count is zero explicitly? And same as before, do we have unit test to ensure leaversResult being null wont cause infinite loop here?

**Review by omio-ma — APPROVED** [review]

File: SchedulerJobs/SchedulerJobs.Services/ELinksService.cs [hunk 4]
```diff
@@ -32,73 +36,91 @@ public ELinksService(IPeoplesClient peoplesClient, ILeaversClient leaversClient,
...
+            var morePages = false;
+            do
             {
-                while (true)
+                try
                 {
                     _logger.LogInformation("ImportJudiciaryPeople: Executing page {CurrentPage}", currentPage);
-                    var people = await _peoplesClient.GetPeopleAsync(fromDate, currentPage);
-                    var peopleResult = people
+                    var peoples = await _peoplesClient.GetPeopleAsync(fromDate, currentPage);
```
Comment (AWildBugAppeared) [inline SchedulerJobs/SchedulerJobs.Services/ELinksService.cs:46]: Not an issue but this should be "people" :D

**Review by AWildBugAppeared — APPROVED** [review]

**Pull request merged**